### DISCUSSION TITLE
Is a custom redirect possible in the component?

### DISCUSSION THREAD

**Discussion by korenevskiy** [joomla/joomla-cms#34734] [General]
Is there such a possibility:
So that I can create a component with a list of products.
But in this list of products, I would indicate a link to the article.
Of course, I will add the prices field to the article.
It is required that when saving the article, the redirect occurs back to the component with the list of products.
This will allow us to use Articles as products in any of its components.

**Comment by chmst** [comment]
If I understand your question correctly: You can define the redirect link in the controller.

**Reply by korenevskiy** [reply]
You misunderstood me.
I'm going to specify a link from one(new) component to another base component (Articles - >Content).
The Articles->Content component is standard for Joomla, I can't change anything in it. since this will be replaced with standard files when updating.
I want to specify a link to the base component Articles->Content from the new component, but at the same time, after saving the data in the base component, a redirect to the first initial component occurs.
You understand that I can't specify a redirect in the Articles->Content controller, since this component will overwrite its files when updating.

**Reply by korenevskiy** [reply]
for example, the URL of the Articles->Content component could have a url for a reverse redirect. When the Articles->Content component sees the redirect parameter in the url, it will redirect the user back to the specified address. And if it does not see this parameter, it will simply open the list of articles of the Articles->Content component after saving the article.

**Reply by brianteeman** [reply]
sounds the same as how j2store works

**Reply by korenevskiy** [reply]
> sounds the same as how j2store works

Yes, exactly like this store. There is a section in the store with a list of products.
After opening the product, we get to the editor of the article. We can put up a price and a photo there. And then we save the article. And then what? We find ourselves in the articles component and we see the articles manager with a list.
But in fact, it is required that after saving the article, we are redirected back to the store component.

**Reply by korenevskiy** [reply]
I want to discuss the function of the reverse redirect to the place from where the transition to the article editor was made.

**Reply by joomdonation** [reply]
If I understand your request right, you can pass the URL you want using return variable (base64_encode)

For example, if you access to this URL https://localhost/j4/administrator/index.php?option=com_content&task=article.add&return=aW5kZXgucGhwP29wdGlvbj1jb21fYmFubmVycw==

After saving the article, you will redirected back to `index.php?option=com_banners`
**aW5kZXgucGhwP29wdGlvbj1jb21fYmFubmVycw==** is result of `base64_encode('index.php?option=com_banners')`
